core/server/master/src/test/java/alluxio/master/AlluxioMasterProcessTest.java

```diff
@@ -179,6 +191,59 @@ public void startMastersThrowsUnavailableException() throws InterruptedException
     Assert.assertTrue(success.get());
   }
 
+  @Test
+  public void failToGainPrimacyWhenJournalCorrupted() throws Exception {
+    Configuration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    URI journalLocation = JournalUtils.getJournalLocation();
+    JournalSystem journalSystem = new JournalSystem.Builder()
+        .setLocation(journalLocation).build(CommonUtils.ProcessType.MASTER);
+    AlluxioMasterProcess masterProcess = new AlluxioMasterProcess(journalSystem);
+    corruptJournalAndStartMasterProcess(masterProcess, journalLocation);
+  }
+
+  @Test
+  public void failToGainPrimacyWhenJournalCorruptedHA() throws Exception {
+    Configuration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    URI journalLocation = JournalUtils.getJournalLocation();
+    JournalSystem journalSystem = new JournalSystem.Builder()
+        .setLocation(journalLocation).build(CommonUtils.ProcessType.MASTER);
+    ControllablePrimarySelector primarySelector = new ControllablePrimarySelector();
+    FaultTolerantAlluxioMasterProcess masterProcess =
+        new FaultTolerantAlluxioMasterProcess(journalSystem, primarySelector);
+    primarySelector.setState(PrimarySelector.State.PRIMARY);
+    corruptJournalAndStartMasterProcess(masterProcess, journalLocation);
+  }
+
+  private void corruptJournalAndStartMasterProcess(AlluxioMasterProcess masterProcess,
+      URI journalLocation) throws Exception {
+    assertTrue(masterProcess.mJournalSystem instanceof UfsJournalSystem);
+    masterProcess.mJournalSystem.format();
+    // corrupt the journal
+    UfsJournal fsMaster =
+        new UfsJournal(URIUtils.appendPathOrDie(journalLocation, "FileSystemMaster"),
+            new NoopMaster(), 0, Collections::emptySet);
+    fsMaster.start();
+    fsMaster.gainPrimacy();
+    long nextSN = 0;
+    try (UfsJournalLogWriter writer = new UfsJournalLogWriter(fsMaster, nextSN)) {
```

Comment:
is it possble to UfsJournal.writer() -> writer.write(the corrupt entry)? instead of creating a new writer?